pkg/ottl/ottlfuncs/func_useragent_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+import (
+	"context"
+	"testing"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	semconv "go.opentelemetry.io/collector/semconv/v1.25.0"
+)
+
+func TestUserAgentParser(t *testing.T) {
+	testCases := []struct {
```

Comment:
Can you add a test case for the Collector's user agent string

Reply:
https://github.com/open-telemetry/opentelemetry-collector/blob/7638bb298172330241375ed5dc3c50637caa645f/exporter/otlphttpexporter/otlp.go#L67

Reply:
Added in commit 7538cf0ecbb006add37328f1e01240c79a463028

Without any modifications the uap-go library parses it as "Other" :sweat_smile: 
If we want to support something specific we can either open a PR in `uap-core` repo (preferred since we can keep using the upstream dependency) or use our own regex file where we add the otel collector regex on top of the existing ones (this would mean we have to sync modifications coming from upstream into out own version of the regular expressions file)

Reply:
If the dep is providing expected user agent strings defined by some standard then I think it is right and we should work to update the collector's user agent format separately.